src/python/pants/backend/project_info/rules/dependencies.py

```diff
@@ -52,22 +53,38 @@ class Dependencies(Goal):
     console: Console, addresses: Addresses, options: DependenciesOptions,
 ) -> Dependencies:
     address_strings: Set[str] = set()
+    third_party_requirements: Set[str] = set()
+
     if options.values.transitive:
         transitive_targets = await Get[TransitiveHydratedTargets](Addresses, addresses)
-        transitive_dependencies = transitive_targets.closure - set(transitive_targets.roots)
-        address_strings.update(
-            hydrated_target.address.spec for hydrated_target in transitive_dependencies
+        hydrated_targets = HydratedTargets(
+            transitive_targets.closure - set(transitive_targets.roots)
         )
     else:
         hydrated_targets = await Get[HydratedTargets](Addresses, addresses)
-        address_strings.update(
-            dep.spec for hydrated_target in hydrated_targets for dep in hydrated_target.dependencies
-        )
-
-    with options.line_oriented(console) as print_stdout:
-        for address in sorted(address_strings):
-            print_stdout(address)
 
+    for target in hydrated_targets:
```

Comment:
For performance reasons, I think that we'll only want to check for 3rdparty and add it to `third_party_requirements` when the `--type` says to do so. Ditto on only checking for `sources` when relevant. You'll probably want to keep this for loop, but tweak the code inside it to first have an if statement for if the `--type` is relevant and then for what type of target you're dealing with.